### PR TITLE
Add snippet for the search_type query parameter

### PR DESCRIPTION
**Pull request by EmmanuelDemey**

### CODE REVIEW

File: docs/reference/search/request/search-type.asciidoc [hunk 1]
```diff
@@ -50,6 +50,11 @@ During the second phase, the coordinating node requests the document
 content (and highlighted snippets, if any) from *only the relevant
 shards*.
 
+[source,js]
+--------------------------------------------------
+GET twitter/_search?search_type=query_then_fetch
+--------------------------------------------------
+
```
Comment (mayya-sharipova) [inline docs/reference/search/request/search-type.asciidoc]: Let's make this example executable/testable by adding `CONSOLE`. This ensures that the snippet will be executed/tested. We will also need to setup the index with data for the test and for this we can use `TEST[setup:twitter]`.

So, the overall snippet can look like this:
```
[source,js]
--------------------------------------------------
GET twitter/_search?search_type=query_then_fetch
--------------------------------------------------
// CONSOLE
// TEST[setup:twitter]
```

File: docs/reference/search/request/search-type.asciidoc [hunk 2]
```diff
@@ -62,4 +67,8 @@ Same as "Query Then Fetch", except for an initial scatter phase which
 goes and computes the distributed term frequencies for more accurate
 scoring.
 
+[source,js]
+--------------------------------------------------
+GET twitter/_search?search_type=dfs_query_then_fetch
+--------------------------------------------------
```
Comment (mayya-sharipova) [inline docs/reference/search/request/search-type.asciidoc]: similar comment here

**Review by mayya-sharipova — CHANGES_REQUESTED** [review]
@EmmanuelDemey Thanks for you contribution! Indeed having examples here would be very helpful. Can you please address comments to ensure the snippets are testable.

**Comment by javanna** [comment]
hi @EmmanuelDemey would you have time to address the comments you got above? Thanks!

**Comment by EmmanuelDemey** [comment]
@javanna sorry for the delay. I have just pushed the change.

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Review by mayya-sharipova — APPROVED** [review]
@EmmanuelDemey Thanks for your PR

**Comment by mayya-sharipova** [comment]
@elasticmachine update branch

**Comment by mayya-sharipova** [comment]
@elasticmachine test this please

**Pull request merged**